java/shared/src/test/java/com/squareup/subzero/shared/SubzeroUtilsTest.java

```diff
@@ -494,6 +503,81 @@ public void testValidateSignTxCommandRequest() {
     assertTrue(e3.getMessage().contains(ERROR_OUTPUTS_COUNT));
   }
 
+  @Test
+  public void testValidateAndSortValidSignatures() {
+    List<ECKey> pubkeys = Arrays.asList(newMockPubKey(), newMockPubKey(), newMockPubKey());
+    pubkeys.sort(ECKey.PUBKEY_COMPARATOR);
+
+    byte[] hash = generateMockHash("mock input");
+      List<Signature> signatures = Arrays.asList(
+        createValidMockSignature(hash, pubkeys.get(0)),
+        createValidMockSignature(hash, pubkeys.get(1))
+    );
+
+    List<byte[]> sortedSigs = SubzeroUtils.validateAndSort(pubkeys, hash, signatures);
+
+    assertEquals(Constants.MULTISIG_THRESHOLD, sortedSigs.size());
+    assertArrayEquals(signatures.get(0).getDer().toByteArray(), sortedSigs.get(0));
+    assertArrayEquals(signatures.get(1).getDer().toByteArray(), sortedSigs.get(1));
+  }
+
+  @Test
+  public void testValidateAndSortInvalidSignatures() {
+    List<ECKey> pubkeys = Arrays.asList(newMockPubKey(), newMockPubKey());
+    byte[] hash = generateMockHash("mock input");
+    List<Signature> signatures = Arrays.asList(
+        createInvalidMockSignature(),
+        createInvalidMockSignature()
+    );
+
+    RuntimeException exception = assertThrows(RuntimeException.class, () ->
+        SubzeroUtils.validateAndSort(pubkeys, hash, signatures)
+    );
+
+    assertTrue(exception.getMessage().contains("Our calculated hash does not match the HSM provided sig"));
+  }
+
+  @Test
+  public void testValidateAndSortSignatureWithWrongHash() {
+    List<ECKey> pubkeys = Arrays.asList(newMockPubKey(), newMockPubKey());
+    byte[] hash = generateMockHash("input1");
+    byte[] wrongHash = generateMockHash("input2");
+    List<Signature> signatures = Arrays.asList(createValidMockSignature(wrongHash, pubkeys.get(0)));
+
+    RuntimeException exception = assertThrows(RuntimeException.class, () ->
+        SubzeroUtils.validateAndSort(pubkeys, hash, signatures)
+    );
+    assertTrue(exception.getMessage().contains("Our calculated hash does not match the HSM provided sig"));
+  }
+
+  @Test
+  public void testValidateAndSortDuplicateSignatures() {
+    List<ECKey> pubkeys = Arrays.asList(newMockPubKey(), newMockPubKey());
+    byte[] hash = generateMockHash("input");
+    Signature validSig = createValidMockSignature(hash, pubkeys.get(0));
+    List<Signature> signatures = Arrays.asList(validSig, validSig);
```

Comment:
nit: what happens if we sign twice with the same key rather than have 2 references to the same sig object? It should fail the same way because bitcoin signatures are deterministic, right? We should probably add that test case too.